src/dbSta/src/dbNetwork.cc

```diff
@@ -4186,4 +4189,123 @@ Net* dbNetwork::getFlatNet(Net* net) const
   return dbToSta(db_net);
 }
 
+void dbNetwork::checkSanityNetConnectivity()
+{
+  // Check for hier net and flat net connectivity
+  dbSet<dbModNet> mod_nets = block()->getModNets();
+  for (auto mod_net : mod_nets) {
+    findRelatedDbNet(mod_net);
+  }
+
+  // Check for incomplete flat net connections
+  for (odb::dbNet* net : block_->getNets()) {
+    if (net->getITerms().size() + net->getBTerms().size() < 2) {
+      logger_->warn(ORD,
+                    2039,
+                    "SanityCheck: Net {} has less than 2 connections.",
+                    net->getName());
+    }
+  }
+}
+
+void dbNetwork::checkSanityModBTerms()
+{
+  if (block_ == nullptr) {
+    return;
+  }
+  for (odb::dbModule* module : block_->getModules()) {
+    std::set<std::string> bterm_names;
+    for (odb::dbModBTerm* bterm : module->getModBTerms()) {
+      const std::string bterm_name = bterm->getName();
+      if (bterm_names.find(bterm_name) != bterm_names.end()) {
```

Comment:
I guess this is ok as a check but I don't see how it could ever happen as dbBTerm::create checks for a duplicate name.  Same for checkSanityModITerms